### PR TITLE
Add support for 32B and 64B swizzles to hopper matmul scheduler

### PR DESCRIPTION
**Pull request by rdspring1**
This PR adds support for 32B and 64B swizzles to StMatrix indexing and to the hopper matmul scheduler.

### Key Index Change
The number of distinct swizzle rows is number of bytes for swizzle divided by size of megabank (16B). The number of times a swizzle pattern is repeated to fill core (8, 8) matrix is number of swizzle rows (8) divided by number of distinct
rows.

```cpp
MmaInputSmemSwizzle swizzle = getSwizzle(out_tv);
int64_t swizzle_bytes = getBytesFromSwizzle(swizzle);
constexpr int64_t megabank_size_bytes = 16;
const int64_t distinct_swizzle_row_size = swizzle_bytes / megabank_size_bytes;

int row = ...;
int col = ...;
constexpr int64_t swizzle_row_size = 8;
const int64_t swizzle_row_repetitions = swizzle_row_size / distinct_swizzle_row_size;
int64_t  row_in_swizzle_pattern = (row % swizzle_row_size) / swizzle_row_repetitions;
int64_t swizzle_col = col ^ row_in_swizzle_pattern;
```

### Testing Changes
* Added `mma_macro` as testing value.
* Created separate test suite called `Swizzle/HopperMatmulSchedulerTest` to test `32B`, `64B`, `128B` swizzles.

### CODE REVIEW

**Comment by rdspring1** [comment]
!test

**Comment by rdspring1** [comment]
!test

File: csrc/scheduler/hopper_multi_matmul.cpp [hunk 1]
```diff
@@ -1027,13 +1027,6 @@ void HopperMultipleMatmulScheduler::scheduleEpilogue() {
     const int64_t tma_m = getM(params_->mma_macro);
     const int64_t tma_n = getN(params_->mma_macro);
 
-    NVF_ERROR(
-        tma_n >= 64,
-        "Scheduler only supports 128B swizzle that requires N dimension of MMA ",
-        "macro to be >= 64, but received ",
-        tma_n,
-        ".");
```
Comment (jacobhinkle) [inline csrc/scheduler/hopper_multi_matmul.cpp:1035]: Do we have restrictions on the size of swizzle anymore? `tma_n` could potentially be any multiple of 8 from 8 to 256. We at least support `tma_n == 64 || tma_n == 32 || tma_n == 16`, but for example do we support `tma_n == 8` or `tma_n == 24`?

**Review by jacobhinkle — APPROVED** [review]
LGTM. I'm just trying to keep track of our constraints. Specifically for internal heuristic development it might be nice to run small instructions like `64, 8, 16` so I wonder if that is supported, or if we have a problem N of 48, is it possible for us.

File: csrc/scheduler/hopper_multi_matmul.cpp (hunk 1 above)
Comment (protonu) [reply csrc/scheduler/hopper_multi_matmul.cpp:1035]: My understanding is we'll support tma_n sizes of 8 or 24 but without any swizzling. Please correct me if I am wrong.

**Review by protonu — APPROVED** [review]

File: csrc/scheduler/hopper_multi_matmul.cpp (hunk 1 above)
Comment (rdspring1) [reply csrc/scheduler/hopper_multi_matmul.cpp:1035]: For now, if the inner dimension is not a multiple of 32, we won't use any swizzling.

File: tests/cpp/test_matmul_scheduler.cpp [hunk 2]
```diff
@@ -3145,6 +3141,15 @@ std::string hopperTestName(
   return os.str();
 }
 
+std::string hopperTestNameSwizzle(
+    const testing::TestParamInfo<HopperMatmulSchedulerTestParams>& info) {
+  std::unordered_map<MmaMacro, std::string> mma_macro_to_swizzle_str_map = {
+      {MmaMacro::Hopper_64_64_16, "128BSwizzle"},
+      {MmaMacro::Hopper_64_32_16, "64BSwizzle"},
+      {MmaMacro::Hopper_64_16_16, "32BSwizzle"}};
```
[the remark this answers is not in the document]
Comment (rdspring1) [reply tests/cpp/test_matmul_scheduler.cpp:3151]: `MmaMacro::Hopper_64_128_16` and `MmaMacro::Hopper_64_256_16` map to `128BSwizzle`

File: tests/cpp/utils.h [hunk 3]
```diff
@@ -703,6 +703,40 @@ static auto kAllHopperMacros = testing::Values(
     MmaMacro::Hopper_64_248_16,
     MmaMacro::Hopper_64_256_16);
 
+static std::unordered_map<MmaMacro, std::string> mma_macro_to_str_map = {
```
[the remark this answers is not in the document]
Comment (rdspring1) [reply tests/cpp/utils.h:706]: I'll do this an a follow-up.

File: csrc/scheduler/hopper_multi_matmul.cpp (hunk 1 above)
Comment (jacobhinkle) [reply csrc/scheduler/hopper_multi_matmul.cpp:1035]: But it is expected to compile. Sounds good.

**Pull request merged**